flink-core/src/main/java/org/apache/flink/api/common/ArchivedExecutionConfig.java

```diff
@@ -44,6 +45,11 @@ public class ArchivedExecutionConfig implements Serializable {
 
     public ArchivedExecutionConfig(ExecutionConfig ec) {
         executionMode = ec.getExecutionMode().name();
+        if (ec.getJobType() != null) {
+            jobType = ec.getJobType();
+        } else {
+            jobType = "UNKNOWN";
```

Comment:
Is it possible for `ec.getJobType()` to be null?

I see `ec.getJobType() = configuration.get(JOB_TYPE);`, and the default value of `JOB_TYPE` is `STREAMING`.